Add notes about API Client

Shadowbeetle opened the pull request:
This Pull Request aims to better emphasize the need to use an API Client when integrating with API adapters.

princemaple commented on lib/swoosh/adapters/amazon_ses.ex:
```diff
@@ -17,6 +17,10 @@ defmodule Swoosh.Adapters.AmazonSES do
         ]
       end
 
+  **All API adapters require an API client.** Swoosh currently supports Hackney and Finch.
```
```suggestion
**This adapter requires an API Client.** Swoosh comes with Hackney and Finch out of the box.
```

Shadowbeetle commented:
Updated the docs in all files that used `Swoosh.ApiClient`. I updated my first commit based on your suggestion,  so it's strange why the `Changes requested` badge is still on.

princemaple commented:
> Updated the docs in all files that used `Swoosh.ApiClient`. I updated my first commit based on your suggestion, so it's strange why the `Changes requested` badge is still on.

You have to click the re-request review button to get rid of it. It's next to the reviewer (me in this case) avatar in the reviewers list.

----

Thanks for the great effort! If there isn't more to be done in this PR, I'm happy to merge it in now.

Shadowbeetle commented:
> You have to click the re-request review button to get rid of it. It's next to the reviewer (me in this case) avatar in the reviewers list.

Thanks! Didn't know that.

> Thanks for the great effort! If there isn't more to be done in this PR, I'm happy to merge it in now.

I think it's ready to be merged. I'm happy I could help a bit!

princemaple commented:
💜
